src/compiler/semantic/mod.rs

```diff
@@ -78,13 +80,32 @@ impl SymTableEntry {
             None => "field",
         }
     }
+
+    fn look_in_usages(
```

Comment:
This method, for self symbol, find its usages that enclose an offset. Right?

perhaps `find_usages_at` ?